dashboard/app/controllers/api/v1/pd/facilitator_program_registrations_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Api::V1::Pd::FacilitatorProgramRegistrationsController < ApplicationController
+  authorize_resource class: 'Pd::FacilitatorProgramRegistration', only: :create
+
+  def create
+    form_data_hash = params.try(:[], :form_data) || {}
+    form_data_json = form_data_hash.to_unsafe_h.to_json.strip_utf8mb4
+
+    teachercon = params.try(:[], :teachercon)
+
+    facilitator_program_registration = ::Pd::FacilitatorProgramRegistration.new(
+      user: current_user,
+      teachercon: teachercon,
+      form_data: form_data_json
+    )
+
+    begin
+      facilitator_program_registration.save
+    rescue
+      # Fail silently - this happens in the event of multiple submission
+      CDO.log.warn "Duplicate pd facilitator program registration submission for user #{current_user.id} - ignoring the second one"
```

Comment:
Maybe it's not right to always fail silently here. In the past we've had teachers who share computers and don't realize that they are signed in to another person's account.

What about failing silently if you have a submission that was just put in very recently, but redirecting to an error/warning otherwise